File: fabric/src/main/java/cc/turtl/chiselmon/fabric/compat/ModMenuIntegration.java
package cc.turtl.chiselmon.fabric.compat;

import cc.turtl.chiselmon.config.ChiselmonConfig;
import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;

public class ModMenuIntegration implements ModMenuApi {
    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return ChiselmonConfig::createScreen;
    }
}

File: fabric/src/main/kotlin/cc/turtl/chiselmon/fabric/client/ModMenuIntegration.kt
package cc.turtl.chiselmon.fabric.client

import cc.turtl.chiselmon.client.config.ChiselmonConfig
import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi

object ModMenuIntegration : ModMenuApi {
    override fun getModConfigScreenFactory(): ConfigScreenFactory<*> {
        return ConfigScreenFactory { parent ->
            ChiselmonConfig.createScreen(parent)
        }
    }
}